plugins/inputs/docker/stats_helpers.go

```diff
@@ -41,8 +41,8 @@ func calculateCPUPercentWindows(v *types.StatsJSON) float64 {
 
 // calculateMemUsageUnixNoCache calculate memory usage of the container.
 // Page cache is intentionally excluded to avoid misinterpretation of the output.
-func calculateMemUsageUnixNoCache(mem types.MemoryStats) float64 {
-	return float64(mem.Usage - mem.Stats["cache"])
+func calculateMemUsageUnixNoCache(mem types.MemoryStats) uint64 {
```

Comment:
@danielnelson may need to weigh in, but I believe changing types isn't supported in certain outputs.

Reply:
That's true but in this case I believe the field types are retained: `usage` is a uint64 and `usage_percent` is a float64.  I think it would be nice to leave this function returning a float64 though so that our code is closer to the upstream code, and then convert to an uint64 just for the usage field.